spec/models/vote_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe Vote, type: :model do
+  it { should validate_presence_of :user_id }
+  it { should validate_inclusion_of(:votable_type).in_array ['Question', 'Answer'] }
+  it { should validate_inclusion_of(:value).in_array [-1, 1] }
+
+  it { should belong_to :user }
+  it { should belong_to :votable }
+
+  let(:users) { create_list(:user, 2) }
+  let(:user) { users.at(0) }
+  let(:other_user) { users.at(1) }
+  let(:question) { create(:question) }
+  let(:answer) { create(:answer, question: question) }
+
+  [:question, :answer].each do |votable_name|
+    before { @votable = send votable_name }
+
+    describe "vote for #{ votable_name }" do
+      it "voted up by user" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.